sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.okhttp;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.Authenticator;
+import okhttp3.ConnectionPool;
+import okhttp3.Dispatcher;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Builder to configure and build an implementation of com.azure.core.http.HttpClient for OkHttp.
+ */
+public class OkHttpAsyncHttpClientBuilder {
+    private final ClientLogger logger = new ClientLogger(OkHttpAsyncHttpClientBuilder.class);
+    private final okhttp3.OkHttpClient okHttpClient;
+    //
+    private final static Duration DEFAULT_READ_TIMEOUT = Duration.ofSeconds(120);
+    private final static Duration DEFAULT_CONNECT_TIMEOUT = Duration.ofSeconds(60);
+    //
+    private List<Interceptor> networkInterceptors = new ArrayList<>();
+    private Duration readTimeout;
+    private Duration connectionTimeout;
+    private ConnectionPool connectionPool;
+    private Dispatcher dispatcher;
+    private java.net.Proxy proxy;
+    private Authenticator proxyAuthenticator;
+
+    /**
+     * Creates OkHttpAsyncHttpClientBuilder.
+     */
+    public OkHttpAsyncHttpClientBuilder() {
+        this.okHttpClient = null;
+    }
+
+    /**
+     * Creates OkHttpAsyncHttpClientBuilder from the builder of an existing OkHttpClient.
+     *
+     * @param okHttpClient the httpclient
+     */
+    public OkHttpAsyncHttpClientBuilder(okhttp3.OkHttpClient okHttpClient) {
+        this.okHttpClient = Objects.requireNonNull(okHttpClient, "okHttpClient == null");
+    }
+
+    /**
+     * Add a network layer interceptor to Http request pipeline.
+     *
+     * @param networkInterceptor the interceptor to add
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder networkInterceptor(Interceptor networkInterceptor) {
+        Objects.requireNonNull(networkInterceptor);
+        this.networkInterceptors.add(networkInterceptor);
+        return this;
+    }
+
+    /**
+     * Add network layer interceptors to Http request pipeline.
+     *
+     * @param networkInterceptors the interceptors to add
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder networkInterceptors(List<Interceptor> networkInterceptors) {
+        this.networkInterceptors = Objects.requireNonNull(networkInterceptors);
+        return this;
+    }
```

Comment:
I think this should be removed.

Reply:
This is to reuse the builder so that caller can clear previous selection by passing empty list or a list with brand new interceptors, any other idea to handle this use case?

Reply:
You should update the JavaDoc to say that this replaces all previously-set interceptors on the builder.